### ISSUE TITLE
Does anyone know why the torso remains without animation in the UE to Rigify addon?

### ISSUE THREAD

**Issue by cyberzond1** [poly-hammer/BlenderTools#96]
Hello everyone!
Does anyone know or can suggest why this "torso" stayed there, while it should have been on the character according to the animation keys? Now it just hangs and doesn't move.

This is the UE to Rigify addon, the latest version, as well as Blender.

In older versions, nothing works at all for me.
Does anyone know why the torso remains without animation in the UE to Rigify addon?
This only happens with the UE5 mannequin :)

If you need any more help or adjustments, feel free to ask!

![image](https://github.com/user-attachments/assets/a45c74d0-a090-4b2e-acd7-0f5707ea1fb8)
![image](https://github.com/user-attachments/assets/cb827703-573f-4697-8b22-5e8e97001546)
I took a Metahuman character or a standard character (for testing) from Unreal 5, exported it to Blender, and applied animation to this character. In the addon, I used the eyedropper tool to select the skeleton, specified the UE5 skeleton, and clicked Convert. An animation rig with animation appeared, but the torso remained without animation. The keys indicate zero coordinates.

I did the same but specified UE4 in the addon. The torso works almost correctly, there is animation. But the skeletons are different, and there is significant distortion in the entire animation.

UE to Rigify 1.7.3
* **Blender Version:** i.e. Blender 4.2.3  3.6
* **Unreal Version:** i.e. Unreal 4.25

I tried a different version of the mod and a different version of Blender. Either the same issue occurs, or it doesn't work at all.
Thank :)

**Comment by cyberzond1** [comment]
[Snachala__5.zip](https://github.com/user-attachments/files/17610760/Snachala__5.zip)
Blender Project

**Comment by JoshQuake** [comment]
Ah thank you. So it sounds like the bones were animated pre-rigging? Usually you want to rig the character first then animate the rig. So because it was animated first, the control rig wont be correct. When animating a rig and baking the animation, the Torso controls transform will basically bake into the Hip bone to be used ingame.

**Comment by cyberzond1** [comment]
Thank you for the response!
Yes, the animation was created. As far as I understand, the add-on's task is not only to convert it into an animation rig but also to save the animation, which can be supplemented or modified. In the videos and examples, everything works. It also works in the UE4 skeleton. But with UE5, it doesn't work at all. I tried on different computers and in different versions, but it doesn't work.

**Comment by JoshQuake** [comment]
Gotcha okay so you're right it is supposed to do that. When you click Convert, make sure Overwrite Animation is ticked. I just tested converting your rig to mannequin 4 template with it ticked and it worked, then I reverted and selected UE5 template and made sure Overwrite Animation was still ticked, and the Torso control is properly following still.

![image](https://github.com/user-attachments/assets/6285ab65-8636-4298-88e7-e7b3dbabafb6)
